src/mango/src/mango_cursor_text.erl

```diff
@@ -334,4 +350,805 @@ get_json_docs(DbName, Hits) ->
         Hits
     ).
 
+%%%%%%%% module tests %%%%%%%%
+
+-ifdef(TEST).
+-include_lib("couch/include/couch_eunit.hrl").
+
+% This behavior needs to be revisited and potentially fixed, the tests
+% here only to record the current version.
+
+create_test_() ->
+    {
+        foreach,
+        fun() ->
+            meck:expect(couch_db, name, [db], meck:val(db_name))
+        end,
+        fun(_) ->
+            meck:unload()
+        end,
+        [
+            ?TDEF_FE(t_create_no_indexes),
+            ?TDEF_FE(t_create_multiple_indexes),
+            ?TDEF_FE(t_create_regular),
+            ?TDEF_FE(t_create_no_bookmark),
+            ?TDEF_FE(t_create_invalid_bookmark)
+        ]
+    }.
+
+t_create_no_indexes(_) ->
+    Exception = {mango_error, mango_cursor_text, multiple_text_indexes},
+    ?assertThrow(Exception, create(db, {[], trace}, selector, options)).
+
+t_create_multiple_indexes(_) ->
+    Indexes = [index1, index2, index3],
+    Exception = {mango_error, mango_cursor_text, multiple_text_indexes},
+    ?assertThrow(Exception, create(db, {Indexes, trace}, selector, options)).
+
+t_create_regular(_) ->
+    Index = #idx{type = <<"text">>},
+    Indexes = [Index],
+    Trace = #{},
+    Limit = 10,
+    Options = [{limit, Limit}, {skip, skip}, {fields, fields}, {bookmark, bookmark}],
+    Options1 = [{limit, Limit}, {skip, skip}, {fields, fields}, {bookmark, unpacked_bookmark}],
+    Cursor = #cursor{
+        db = db,
+        index = Index,
+        ranges = null,
+        trace = Trace,
+        selector = selector,
```

Comment:
A selector here obviously doesn't matter and can be anything, however in general in tests, if possible it might be better use proper types even when they don't matter and when it's not too hard. So maybe a selector might be `{[]}`. `skip` should be perhaps 0 or some plausible integer like `53` etc. Don't have to change in the current PR just in general I think it's nicer to respect types.